stdlib/public/Cxx/CxxRandomAccessCollection.swift

```diff
@@ -36,16 +36,52 @@ extension CxxRandomAccessCollection {
     return Int(__endUnsafe() - __beginUnsafe())
   }
 
+  @inlinable
+  @inline(__always)
+  internal func _getRawIterator(at index: Int) -> RawIterator {
+    var rawIterator = self.__beginUnsafe()
+    rawIterator += RawIterator.Distance(index)
+    precondition(self.__endUnsafe() - rawIterator > 0,
```

Comment:
What is the plan for these when hardening for libcxx is on? Are we relying on the optimizer to remove the redundant checks or do we have a way to disable precondition checks when hardening is turned on?

Reply:
@susmonteiro might know more details here, but the general idea is that there won't be a bounds-check here with hardening enabled, _unless_ you also enable libc++ bounded iterators. This is because we're not using the C++ `operator[]` – this function would only be invoked if the C++ type does not have a suitable overload of `operator[]`.

We don't currently have a good way to check if bounded iterators are enabled and provide an alternative implementation, so for now we're just hoping that the optimizer removes these checks, I think.